### PR TITLE
feat(Mixpanel):: Revamping code.

### PR DESCRIPTION
**Pull request by ItsSudip**
## Description of the change

We are revamping the flow of `identify` call, adding support of `/import` endpoint by default for track calls and also refactoring the code.

## Type of change
- [X] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [X] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [X] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/mp/transform.js [hunk 1]
```diff
@@ -1,101 +1,141 @@
...
   getErrorRespEvents,
   getSuccessRespEvents,
   CustomError,
-  isAppleFamily,
-  getFullName,
-  extractCustomFields
+  isHttpStatusSuccess
 } = require("../../util");
+const { ConfigCategory, mappingConfig } = require("./config");
+
```
Comment (utsabc) [inline v0/destinations/mp/transform.js:18]: Remove these line spaces between imports

File: v0/destinations/mp/transform.js [hunk 2]
```diff
@@ -158,107 +197,33 @@ function processTrack(message, destination) {
...
+  // If userId and anonymousId both are present and required credentials for /import
+  // endpoint are available we are creating the merging response below
+  // https://developer.mixpanel.com/reference/identity-merge
+  if (
+    message.userId &&
+    message.anonymousId &&
+    (destination.Config.apiSecret ||
+      (destination.Config.serviceAccountSecret &&
+        destination.Config.serviceAccountUserName &&
+        destination.Config.projectId))
```
Comment (utsabc) [inline v0/destinations/mp/transform.js:221]: The config credentials related presence validation can be move to a mico function with proper name for ease of understanding

File: v0/destinations/mp/transform.js [hunk 3]
```diff
@@ -1,101 +1,141 @@
...
-          ).toString("base64")}`
-        };
-      } else {
-        throw new CustomError(
-          "Event timestamp is older than 5 days and no apisecret is provided in destination config.",
-          400
-        );
-      }
-    }
+async function createUser(response) {
```
Comment (utsabc) [inline v0/destinations/mp/transform.js:35]: Better add documentation for this function

File: v0/destinations/mp/transform.js [hunk 4]
```diff
@@ -1,101 +1,141 @@
...
+  const url = response.endpoint;
+  const { params } = response;
+  const axiosResponse = await httpPOST(url, {}, { params });
+  const processedResponse = processAxiosResponse(axiosResponse);
+  if (processedResponse.response === 1) {
+    return processedResponse.status;
   }
+  return 400;
```
Comment (utsabc) [inline v0/destinations/mp/transform.js:43]: What if `processedResponse` has got a 5XX from mp iteself shouldn't we return that?

File: v0/destinations/mp/transform.js [hunk 5]
```diff
@@ -1,101 +1,141 @@
...
+}
 
-  const encodedData = Buffer.from(
-    JSON.stringify(removeUndefinedValues(parameters))
-  ).toString("base64");
+function setImportCredentials(destConfig) {
+  const endpoint =
+    destConfig.dataResidency === "eu"
+      ? "https://api-eu.mixpanel.com/import/"
+      : "https://api.mixpanel.com/import/";
```
Comment (utsabc) [inline v0/destinations/mp/transform.js:50]: We can store this in Constants

File: v0/destinations/mp/transform.js [hunk 6]
```diff
@@ -1,101 +1,141 @@
...
+  } = destConfig;
+  if (apiSecret) {
+    headers.Authorization = `Basic ${Buffer.from(`${apiSecret}:`).toString(
+      "base64"
+    )}`;
+  } else if (serviceAccountUserName && serviceAccountSecret && projectId) {
+    headers.Authorization = `Basic ${Buffer.from(
+      `${serviceAccountUserName}:${serviceAccountSecret}`
+    ).toString("base64")}`;
+    params.projectId = projectId;
```
Comment (utsabc) [inline v0/destinations/mp/transform.js:67]: A micor function would simplyfy things here

File: v0/destinations/mp/transform.js (hunk 1 above)
Comment (ItsSudip) [reply v0/destinations/mp/transform.js]: Updated

File: v0/destinations/mp/transform.js (hunk 2 above)
Comment (ItsSudip) [reply v0/destinations/mp/transform.js]: Updated

File: v0/destinations/mp/transform.js (hunk 3 above)
Comment (ItsSudip) [reply v0/destinations/mp/transform.js:45]: Updated

File: v0/destinations/mp/transform.js (hunk 4 above)
Comment (ItsSudip) [reply v0/destinations/mp/transform.js]: Updated

File: v0/destinations/mp/transform.js (hunk 5 above)
Comment (ItsSudip) [reply v0/destinations/mp/transform.js]: Updated

File: v0/destinations/mp/transform.js (hunk 6 above)
Comment (ItsSudip) [reply v0/destinations/mp/transform.js]: Updated

**Review by utsabc — DISMISSED** [review]

**Review by arnab-p — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**